### PR TITLE
Further type fixes

### LINKED ISSUE #13656: [Task]: Fix possible errors after applying php native types
### Improvement description

Follow up to #12240

### PR DESCRIPTION
**Pull request by fashxp**
followup to #13653

### TODO
- [x] add relevant method signature changes to that list: https://github.com/pimcore/pimcore/issues/13627

### CODE REVIEW

File: models/DataObject/ClassDefinition/Data/Block.php [hunk 1]
```diff
@@ -554,7 +554,10 @@ public function hasChildren(): bool
         return false;
     }
 
-    public function addChild(Data|Layout $child)
+    /**
+     * @param Data|Layout $child
+     */
+    public function addChild(mixed $child)
```
Comment (fashxp) [inline models/DataObject/ClassDefinition/Data/Block.php:563]: @dvesh3 should we add a comment here (and all the other places) why `mixed` is necessary?

Comment (dvesh3) [reply models/DataObject/ClassDefinition/Data/Block.php:563]: yes, this is required for assets metadata definitions bundle. please see https://github.com/pimcore/asset-metadata-class-definitions/actions/runs/3576806963/jobs/6015056604#step:10:47

I'll add comments here once the approach is finalized.

File: lib/Http/ResponseHelper.php [hunk 2]
```diff
@@ -61,7 +61,7 @@ public function isHtmlResponse(Response $response): bool
             return false;
         }
 
-        if (false !== strpos($response->getContent(), '<html')) {
+        if (false !== strpos((string)$response->getContent(), '<html')) {
```
Comment (brusch) [inline lib/Http/ResponseHelper.php:64]:
```suggestion
        if (false !== strpos($response->getContent(), '<html')) {
```

See #13710

**Review by brusch — DISMISSED** [review]

File: lib/Http/ResponseHelper.php (hunk 2 above)
Comment (fashxp) [reply lib/Http/ResponseHelper.php:64]: lets keep the cast too ...

File: models/DataObject/ClassDefinition/Data/Block.php (hunk 1 above)
Comment (dvesh3) [reply models/DataObject/ClassDefinition/Data/Block.php:563]: done here https://github.com/pimcore/pimcore/pull/13705/commits/cbbbe8150108d3b2a1387eae678e1bd103998dc4

**Pull request merged**